commit 4bf97c196aba9e50289333c15336ad109fc87f2d
Author: Anatol Pomozov <anatol.pomozov@gmail.com>
Date:   Mon Nov 24 11:12:52 2014 -0800

    Do not drop CFLAGS from c_glib compilation
    
    CFLAGS contains important flags. Dropping it introduces e.g. such issues:
    https://sourceware.org/bugzilla/show_bug.cgi?id=13979
    
    Signed-off-by: Anatol Pomozov <anatol.pomozov@gmail.com>
    
    This closes #284

diff --git a/lib/c_glib/Makefile.am b/lib/c_glib/Makefile.am
index 88e2d73..4e0aa88 100755
--- a/lib/c_glib/Makefile.am
+++ b/lib/c_glib/Makefile.am
@@ -24,11 +24,7 @@ pkgconfigdir = $(libdir)/pkgconfig
 lib_LTLIBRARIES = libthrift_c_glib.la
 pkgconfig_DATA = thrift_c_glib.pc
 
-common_cflags = -g -Wall -W -Isrc -I src/thrift/c_glib $(GLIB_CFLAGS)
-common_ldflags = -g -Wall -W $(GLIB_LDFLAGS) @GCOV_LDFLAGS@
-
-# this removes optimizations and adds coverage flags
-CFLAGS = @GCOV_CFLAGS@
+AM_CFLAGS = -Isrc -I src/thrift/c_glib
 
 # Define the source files for the module
 
@@ -54,7 +50,7 @@ libthrift_c_glib_la_SOURCES = src/thrift/c_glib/thrift.c \
                               src/thrift/c_glib/server/thrift_server.c \
                               src/thrift/c_glib/server/thrift_simple_server.c
 
-libthrift_c_glib_la_CFLAGS = $(common_cflags)
+libthrift_c_glib_la_CFLAGS = $(AM_CFLAGS) $(GLIB_CFLAGS)
 
 include_thriftdir = $(includedir)/thrift/c_glib
 include_thrift_HEADERS = \
